db/version_set.cc

```diff
@@ -2028,7 +2028,6 @@ void VersionStorageInfo::ComputeBottommostFilesMarkedForCompaction() {
   bottommost_files_mark_threshold_ = kMaxSequenceNumber;
   for (auto& level_and_file : bottommost_files_) {
     if (!level_and_file.second->being_compacted &&
-        level_and_file.second->fd.largest_seqno != 0 &&
```

Comment:
Ideally `level_and_file.second->fd.largest_seqno != 0` tells us that a file has at least one of these properties:

- contains overwritten keys
- contains deleted keys
- contains range deletions

Prior to your change it definitely didn't tell us that because the compaction's final key seqnum wasn't zeroed out. After your change it might be enough, though we should test (I think I saw once a compaction's first key seqnum not being zeroed... we need to make sure that doesn't happen)

Reply:
The issues with using `level_and_file.second->num_deletions > 1` instead are:

- it doesn't trigger compaction when there's one deletion, which could be a range deletion covering a lot of keys in the file
- it doesn't trigger compaction when there are many overwritten keys. Some MyRocks use case actually doesn't use `Delete` but instead calls `Put` with empty value to "delete" keys, so we'd like to be able to trigger compaction in this case too.